src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp

```diff
@@ -237,55 +261,77 @@ class boss_archimonde : public CreatureScript
         }
 
         InstanceScript* instance;
+        EventMap events;
 
         uint64 DoomfireSpiritGUID;
         uint64 WorldTreeGUID;
 
-        uint32 DrainNordrassilTimer;
-        uint32 FearTimer;
-        uint32 AirBurstTimer;
-        uint32 GripOfTheLegionTimer;
-        uint32 DoomfireTimer;
-        uint32 SoulChargeTimer;
         uint8 SoulChargeCount;
-        uint32 MeleeRangeCheckTimer;
-        uint32 HandOfDeathTimer;
-        uint32 SummonWispTimer;
         uint8 WispCount;
-        uint32 EnrageTimer;
-        uint32 CheckDistanceTimer;
 
         bool Enraged;
         bool BelowTenPercent;
         bool HasProtected;
         bool IsChanneling;
 
+        std::list<Unit*> targets;
```

Comment:
Why is this list scoped at class level? It could introduce bugs since it's being used by different functions for different events (and is not being cleared).